src/main/java/view/CitiBikeController.java

```diff
@@ -0,0 +1,51 @@
+package view;
+
+import logic.ClosestStationFinder;
+import model.StationInformation;
+import model.StationStatus;
+import org.jxmapviewer.viewer.GeoPosition;
+import service.CitiBikeService;
+import service.CitiBikeServiceFactory;
+
+public class CitiBikeController {
+    private final CitiBikeComponent view;
+    private final CitiBikeService service;
+
+    public CitiBikeController(CitiBikeComponent view) {
+        this.view = view;
+        CitiBikeServiceFactory factory = new CitiBikeServiceFactory();
+        this.service = factory.getService();
+    }
+
+    public void calculateRoute() {
+        if (view.getFromPosition() != null && view.getToPosition() != null) {
+            StationInformation stationInfo = service.getStationInformation().blockingGet();
+            StationStatus stationStatus = service.getStationStatus().blockingGet();
```

Comment:
You cannot use blockingGet in UI.

Also you're supposed to be using your lambda for this.

Reply:
how do we use lambda for this

Reply:
It's running in the cloud.

It does the work you want to do.

You know how to make requests and get responses...

Reply:
disposable = data.subscribeOn(Schedulers.io())
                .observeOn(SwingSchedulers.edt())
                .subscribe(
                        this::handleResponse,
                        Throwable::printStackTrace); do we have to do something like this?

Reply:
Yes